flytekit/types/numpy/ndarray.py

```diff
@@ -11,6 +13,17 @@
 from flytekit.models.types import LiteralType
 
 
+def extract_metadata(t: Type[np.ndarray]) -> Tuple[Type[np.ndarray], Dict[str, bool]]:
+    metadata = None
```

Comment:
We can simplify uses of `metadata` if we initialize `metadata` as an empty dictionary (i.e. we won't need to always check if it's non-null).

Reply:
Ah, you're right!